pkg/publicapi/binder.go

```diff
@@ -0,0 +1,33 @@
+package publicapi
+
+import (
+	"github.com/labstack/echo/v4"
+)
+
+// normalizable is an interface that defines a Normalize method.
+type normalizable interface {
+	Normalize()
+}
+
+// CustomBinder is a custom binder that extends the default binder with normalization.
+type CustomBinder struct {
```

Comment:
How about `NormalizeBinder`

Reply:
Renamed in https://github.com/bacalhau-project/bacalhau/pull/4136/commits/8151aad82b3c72545f6fb1cf06fb13b49549ece8